includes/multi-currency/class-currency-switcher-widget.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * WooCommerce Payments Currency Switcher Widget
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Multi_Currency;
+
+use WP_Widget;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Currency Switcher Widget Class
+ */
+class Currency_Switcher_Widget extends WP_Widget {
+
+	const DEFAULT_SETTINGS = [
+		'title'  => '',
+		'symbol' => true,
+		'flag'   => false,
+	];
+
+	/**
+	 * Multi-Currency instance.
+	 *
+	 * @var Multi_Currency
+	 */
+	protected $multi_currency;
+
+	/**
+	 * Register widget with WordPress.
+	 *
+	 * @param Multi_Currency $multi_currency The Multi_Currency instance.
+	 */
+	public function __construct( Multi_Currency $multi_currency ) {
+		$this->multi_currency = $multi_currency;
+
+		parent::__construct(
+			'currency_switcher_widget',
+			__( 'Currency Switcher', 'woocommerce-payments' ),
+			[ 'description' => __( 'Let your customers switch between your enabled currencies', 'woocommerce-payments' ) ]
+		);
+	}
+
+	/**
+	 * Front-end display of widget.
+	 *
+	 * @param array $args     Widget arguments.
+	 * @param array $instance Saved values from database.
+	 */
+	public function widget( $args, $instance ) {
+		$instance = wp_parse_args(
+			$instance,
+			self::DEFAULT_SETTINGS
+		);
+
+		$title = apply_filters( 'widget_title', $instance['title'] );
```

Comment:
Do we need to add a filter to the title here? I mean, merchants would already be able to change it from the widget settings, do we need to change it from somewhere else?